### PR TITLE
feat: modify events to return more information about entity during auditing

### PR DESCRIPTION
**Pull request by yquansah**
For audit events, we previously were not returning important information about the entity due to only logging the request payload.

In this PR, we are changing the structure to include an audit representation of the entities for sending out to the different configured sinks.

### CODE REVIEW

File: internal/server/middleware/grpc/middleware_test.go [hunk 1]
```diff
@@ -957,7 +957,7 @@ func TestAuditUnaryInterceptor_DeleteVariant(t *testing.T) {
 	assert.NotNil(t, got)
 
 	span.End()
-	assert.Equal(t, exporterSpy.GetSendAuditsCalled(), 1)
+	assert.Equal(t, 1, exporterSpy.GetSendAuditsCalled())
```
Comment (yquansah) [inline internal/server/middleware/grpc/middleware_test.go:960]: I realized I had the order on these switched up (expected vs. actual).

File: internal/server/audit/audit.go [hunk 2]
```diff
@@ -31,14 +32,14 @@ type Type string
...
-	Token        Type = "token"
-	Variant      Type = "variant"
+	ConstraintType   Type = "constraint"
+	DistributionType Type = "distribution"
+	FlagType         Type = "flag"
+	NamespaceType    Type = "namespace"
+	RuleType         Type = "rule"
+	SegmentType      Type = "segment"
+	TokenType        Type = "token"
+	VariantType      Type = "variant"
```
Comment (yquansah) [inline internal/server/audit/audit.go:42]: I created the actual entities for the `audit` package. I was at a crossroads when trying to come up with a name for these types. Let me know what y'all think?

**Comment by yquansah** [comment]
The diff looks a little strange for the `Delete` operations. I had to `switch` on the gRPC request for those, because they all return an empty response. For creates, and updates, they both return the actual entity.

Maybe there is a better way to do it? Let me know what y'all think.

File: internal/server/audit/audit.go (hunk 2 above)
Comment (markphelps) [reply internal/server/audit/audit.go:42]: I dig it

File: internal/server/audit/types.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	FlagKey      string `json:"flag_key"`
+	Key          string `json:"key"`
+	Name         string `json:"name"`
+	Description  string `json:"description"`
+	Attachment   string `json:"attachment"`
+	NamespaceKey string `json:"namespace_key"`
+}
+
+func NewVariant(v *flipt.Variant) *Variant {
+	return &Variant{
```
Comment (markphelps) [inline internal/server/audit/types.go:39]: could we add some very simple tests for these `NewX` funcs? Would bump up our test coverage a bit.

Comment (yquansah) [reply internal/server/audit/types.go:39]: @markphelps Doing that now!

**Review by GeorgeMac — APPROVED** [review]
Nice 💪

File: internal/server/audit/types_test.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	}
+
+	nv := NewVariant(v)
+	assert.Equal(t, nv.Id, v.Id)
+	assert.Equal(t, nv.FlagKey, v.FlagKey)
+	assert.Equal(t, nv.Key, v.Key)
+	assert.Equal(t, nv.Name, v.Name)
+}
+
+func testConstraintHelper(t *testing.T, c *flipt.Constraint) {
```
Comment (markphelps) [inline internal/server/audit/types_test.go:40]: nit: could we add `t.Helper()` as first lines in all these helper methods? https://pkg.go.dev/testing#T.Helper

**Review by markphelps — COMMENTED** [review]
one minor nit, otherwise looks great to me!

File: internal/server/audit/types_test.go (hunk 4 above)
Comment (yquansah) [reply internal/server/audit/types_test.go:40]: @markphelps Cool will do that now!

**Pull request merged**